packages/mock-block-dock/src/useMockDatastore.ts

```diff
@@ -176,7 +180,92 @@ export const useMockDatastore = (
       );
       return new Array(actions.length).fill(true);
     },
-    [],
+    [setEntities],
+  );
+
+  const createEntityTypes: BlockProtocolCreateEntityTypesFunction = useCallback(
+    async (actions) => {
+      const newEntityTypes = actions.map<BlockProtocolEntityType>((action) => {
+        const entityTypeId = uuid();
+        const { accountId, schema } = action;
+        return {
+          accountId,
+          entityTypeId,
+          ...schema,
+        } as BlockProtocolEntityType; // @todo fix this – make 'schema' a narrower object in BP types?
+      });
+      setEntityTypes((currentEntityTypes) => [
+        ...currentEntityTypes,
+        ...newEntityTypes,
+      ]);
+      return newEntityTypes;
+    },
+    [setEntityTypes],
+  );
+
+  const getEntityTypes: BlockProtocolGetEntityTypesFunction = useCallback(
+    async (actions) =>
+      actions.map((action) => {
+        const foundEntityType = entityTypes.find((entityType) =>
+          matchEntityTypeIdentifiers(entityType, action),
+        );
+        if (!foundEntityType) {
+          throw new Error(
+            `Could not find entity type for getEntityTypes action ${JSON.stringify(
+              action,
+            )}`,
+          );
+        }
+        return foundEntityType;
+      }),
+    [entityTypes],
+  );
+
+  const updateEntityTypes: BlockProtocolUpdateEntityTypesFunction = useCallback(
+    async (actions) => {
+      const updatedEntityTypes: BlockProtocolEntityType[] = [];
+      setEntityTypes((currentEntityTypes) =>
+        currentEntityTypes.map((entityType) => {
+          const actionToApply = actions.find((action) =>
+            matchEntityTypeIdentifiers(action, entityType),
+          );
+          if (actionToApply) {
+            const newEntityType = {
+              ...actionToApply.schema,
+              accountId: entityType.accountId,
+              entityTypeId: entityType.entityTypeId,
+            } as BlockProtocolEntityType;
+            updatedEntityTypes.push(newEntityType);
+            return newEntityType;
+          }
+          return entityType;
+        }),
+      );
+      return updatedEntityTypes;
+    },
+    [setEntityTypes],
+  );
+
+  const deleteEntityTypes: BlockProtocolDeleteEntityTypesFunction = useCallback(
+    async (actions) => {
+      setEntityTypes((currentEntityTypes) =>
+        currentEntityTypes
+          .map((entityType) => {
+            const isMatch = actions.some((action) =>
+              matchEntityTypeIdentifiers(entityType, action),
+            );
+            if (isMatch) {
+              return null;
+            }
+            return entityType;
+          })
+          .filter(
+            (entityType): entityType is BlockProtocolEntityType => !!entityType,
+          ),
```

Comment:
Not sure if I'm missing something obvious, but why not just:
```suggestion
        currentEntityTypes
          .filter(
            (entityType) => !actions.some((action) =>
              matchEntityTypeIdentifiers(entityType, action),
            ),
          ),
```
So that you can implement the boolean return array more easily down the line?

Reply:
I think I just copied this code from other methods and didn't think about it properly.

I've done the boolean return properly [here](https://github.com/blockprotocol/blockprotocol/pull/302/files/618ce2a25736a0f8a9c38acaab9bf6fdac2486d5..c2a41e205081a3630491aae8f5720d9b2edd5323)